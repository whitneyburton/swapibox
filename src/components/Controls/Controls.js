import React, { Component } from 'react';
import './Controls.scss';

export class Controls extends Component {
  constructor(props) {
    super(props);
    this.state = {
      active: false,
    }
  }
  
  passCategory = (e) => {
    let { retrieveCategory } = this.props;
    const category = e.target.name;
    retrieveCategory(category);
  }

  render() {
    const favorites = this.props.favorites.length;
    const categories = ['PEOPLE', 'PLANETS', 'VEHICLES', `FAVORITES ${favorites}`]
    const buttons = categories.map(category => {
      return <button
        onClick={this.passCategory}
        type='submit'
        name='people'
        className='people-btn btns'>${category}</button>
    })
    return (
      <div className='Controls'>
        <button
          onClick={this.passCategory}
          type='submit'
          name='people'
          className='people-btn btns'>PEOPLE</button>
        <button
          onClick={this.passCategory}        
          type='submit'
          name='planets'
          className='planets-btn btns'>PLANETS</button>
        <button
          onClick={this.passCategory}        
          type='submit'
          name='vehicles'
          className='vehicles-btn btns'>VEHICLES</button>
        <button
          onClick={this.passCategory}        
          type='submit'
          name='favorites'
          className='favorites-btn btns'>FAVORITES (0)</button>
      </div>
    )
  }
}