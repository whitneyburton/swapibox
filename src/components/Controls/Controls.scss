.Controls {
  background-color: transparent;
  height: 100%;
  grid-area: controls;
  display: flex;
  justify-content: space-around;
  align-items: center;
  padding-top: 10px;
}

.people,
.planets,
.vehicles {
  border-radius: 5px;
  padding: 5px 10px;
  background-color: #0013ff69;
  border: none;
  color: #fff;

  &:hover {
    cursor: pointer;
  }
}

.favorites {
  border-radius: 5px;
  padding: 5px 10px;
  background-color: #ffd811c4;
  font-size: 14px;
  border: none;
  color: #fff;

  &:hover {
    cursor: pointer;
  }
}