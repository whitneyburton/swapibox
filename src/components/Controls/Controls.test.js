import React from 'react';
import { Controls } from './Controls';
import { shallow } from 'enzyme';

describe('Controls', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Controls />)
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})