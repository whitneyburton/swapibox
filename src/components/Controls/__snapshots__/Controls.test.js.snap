// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should match the snapshot 1`] = `
<div
  className="Controls"
>
  <button
    className="people-btn btns"
    name="people"
    onClick={[Function]}
    type="submit"
  >
    PEOPLE
  </button>
  <button
    className="planets-btn btns"
    name="planets"
    onClick={[Function]}
    type="submit"
  >
    PLANETS
  </button>
  <button
    className="vehicles-btn btns"
    name="vehicles"
    onClick={[Function]}
    type="submit"
  >
    VEHICLES
  </button>
  <button
    className="favorites-btn btns"
    name="favorites"
    onClick={[Function]}
    type="submit"
  >
    FAVORITES (0)
  </button>
</div>
`;
